### ISSUE TITLE
Network fails to converge

### ISSUE THREAD

**Issue by dtmoodie** [pjreddie/darknet#186]
Hello,

I've trained several YOLO models using 0710531b666d233cae5f88706c754d900a2600af
I recently updated to master/HEAD and now the same training scripts diverge to NaN  before the burn in period finishes.
I will be looking into which commit specifically causes the divergence, but I thought it would be pertinent to let people know.

**Comment by HencyChen** [comment]
Try to tune the training subvision to 16 and see if it works

**Comment by dtmoodie** [comment]
Would that be an expected lowering or raising of the subdivisions?  What does that parameter do?
I previously had to use a batch size of 4 and subdivision of 1 otherwise I would run out of memory when training with 4 GPUs.  I've since switched to one GPU where I can train with a batch size of 64 with 16 subdivisions however this still starts to diverge after 2000 iterations.

**Comment by yutingchen1211** [comment]
Using a smaller learning rate?

**Comment by dtmoodie** [comment]
I've toyed with learning rate and found it to be ineffective. 
After significant debugging, I've found that YOLO doesn't like some of the data that I'm passing into it, particularly images of mostly snow.  I found that using a smooth L1 loss instead of L2 loss helps make it more robust to these images but this still isn't sufficient to guarantee convergence.  I've also found that it is particularly vulnerable to these images if the class scale is higher than 1.
